api/clients/v2/verification/cert_verifier.go

```diff
@@ -245,3 +247,40 @@ func (cv *CertVerifier) getVerifierCallerFromAddress(
 	cv.verifierCallers.Store(certVerifierAddress, certVerifierCaller)
 	return certVerifierCaller, nil
 }
+
+// GetConfirmationThreshold returns the ConfirmationThreshold that corresponds to the input reference block number.
+//
+// This method will return the confirmation threshold from an internal cache if it is already known for the cert
+// verifier which corresponds to the input reference block number. Otherwise, this method will query the confirmation
+// threshold and cache the result for future use.
```

Comment:
this cache will be useless afaiu because most blobs have different RBN. A better cache would be to have a list of blocks at which teh certVerifier were updated, and then when you get an RBN you check which certVerifier it belongs to (by scanning the list of update blocks and checking where RBN lands).

Not sure its worth implementing, but maybe just worth adding a comment to that effect.

Reply:
It's not caching based on RBN, it's caching based on cert verifier address. From the doc:

> ... if it is already known for the **cert verifier** which corresponds to the input reference block number.

RBN is the parameter passed in, since it's required to determine which cert verifier address is active.

LMK if the logic here is still unclear, but AFAICT it's implemented correctly, and will have a practical impact (so that we only need to fetch confirmation threshold once for each time the cert verifier address changes)

Reply:
Ahhh gotcha! Yep my bad, this is good